blog/docs/articles/llm-agents-demo.md

```diff
@@ -0,0 +1,136 @@
+
+In the previous blog post on LLM Agents, we discussed what LLM Agents are,
+how LLMs call tools, and how Knative can be used to automate the tool calling
+process for LLM Agents. In this blog post we will be looking at a concrete
+example of how you can build an LLM Agent application with Knative handling
+the tool calling and discovery, and explaining what differentiates this
+approach from the other common approaches for building LLM Agents.
+
+## An example LLM Agent using Knative
+
+To showcase the power of building LLM Agents with the tool discovery and
+calling handled by Knative, we are going to build a simple chat application
+that is capable of answering questions about the cost of resources being
+used for a fictitious application we are running. Specifically, in our
+application we are interested in accurately answering questions such as:
+
+> How has the trend in the average resource consumption (CPU, memory)
+changed over the past few months for my application, and what impact
+will this have on my costs?
+
+To begin, we will be deploying the Knative LLM Tool Provider demo into
+a cluster we have access to following the instructions in the README in
+https://github.com/keventmesh/llm-tool-provider. Once we have deployed
+the chat app from this repository, we are able to access it by running:
+
+```sh
+kubectl port-forward svc/chat-app-service 8080:8080
+```
+
+If you access http://localhost:8080 in your browser you will now be able
+to see the chat application running, and ask it our simple question. When
+I tried this while writing the blog post, the interaction was:
+
+Calum:
+> How has the trend in the average resource consumption (CPU, memory)
+changed over the past few months for my application, and what impact
+will this have on my costs?
+
+LLM Agent:
+> Could you provide me with the data on the average resource consumption
+(CPU, GPU, memory) for your application over the past few months?
+
+If you try this you will likely notice that your LLM agent replies similarly,
+and that the UI will indicate in the task list a task called “human”. This is
+the LLM calling the one hand-coded tool that we have in our sample app: a tool
+that asks the user (the human) for more information before continuing to answer
+the question.
+
+From this interaction, we can see that the LLM isn’t capable of answering our
+question yet: it has no information about the resource consumption or about
+what the costs associated with that resource consumption are. To give the LLM
+these capabilities we need to tell it about the two Knative Functions we
+deployed into our cluster in the setup of the sample app:
+i`average-resource-consumption` and `resource-cost-calculator`. To do this
+we will need to apply two EventTypes into our cluster which describe the
+contracts expected by these Functions:
+
+```yaml
+apiVersion: eventing.knative.dev/v1beta2
+kind: EventType
```

Comment:
When I read to here, I become curious on how the LLM knows these eventTypes are for the Agent to use.